### ISSUE TITLE
کلمات هم واژه

### ISSUE THREAD

**Issue by rayman3003** [sir-kokabi/Vajehdan#2]
خسته نباشید

یه پیشنهادی داشتم برای این برنامه ی خوبتون

اگه امکانش هست, لطفی یه قسمتی به واژه دان اضافه کنید (مثلا با عنوانِ "هم واژه") تا ما با وارد کردنِ یه کلمه, بتونیم تمامِ کلماتی که با اون کلمه شروع میشن, و یا به اون کلمه ختم میشن رو هم داشته باشیم.

 برای مثال:

با وارد کردنِ کلمه ی "بان", کلماتِ زیر نمایش داده بشن:

-شروع شونده ها-
بانداژ - باند - بانه - بانو - بانک - ...

-پایان شونده ها-
نگهبان - باغبان - دربان - قربان - ...

**Comment by ayub-kokabi** [comment]
سلام.
ممنون از پیشنهادتون.
ویژگی بسیار خوبیه. گاهی پسوند با پیشوند کلمه یادمونه ولی اصل کلمه نه. 

> چی چی مال بود؟

 مال رو سرچ می‌کنم، تیک دو گزینه رو می‌زنم «مالامال» رو پیدا می‌کنم.

به نظرتون طراحی پایین، این نیاز رو به خوبی مطرح می‌کنه؟ 
حالت‌های دیگری نیست که پوشش بدیم؟

![image](https://github.com/sir-kokabi/Vajehdan/assets/116794136/ba2335f3-7570-4519-af2d-e7544b0cbe7f)


پی‌نوشت ــــــــــــــــ
واژه‌دان در حال حاضر تحت توسعه فعال نیست؛ اگر فرصتی فراهم شد این ویژگی رو اضافه می‌کنم.
ممنون برای معرفی قابلیت.

**Comment by rayman3003** [comment]
> به نظرتون طراحی پایین، این نیاز رو به خوبی مطرح می‌کنه؟

تشکر. بله. خوب بنظر میرسه

> حالت‌های دیگری نیست که پوشش بدیم؟

فکر کنم یه حالتِ دیگه هم هست. یعنی "عبارت در وسط"

 مثال برای همون "بان": تاباندن - تبانی - نگهبانی - ...

کمتر از "آغازین" و "پایانی" کاربردیه, ولی باز هم شاید برای بعضی از کاربرها مهم باشه. پس اگه این "عبارت در وسط" هم اضافه کنید, دیگه فکر کنم بی نقص بشه D:

> اگر فرصتی فراهم شد این ویژگی رو اضافه می‌کنم.

ممنون. هر موقع فرصتی براتون پیش اومد, این قابلیتو اضافه کنید. و اینجا منشن کنید, و من تستش میکنم. تشکر

**Comment by rayman3003** [comment]
@ayub-kokabi 
میخواستم بدونم آیا همچنان فرصتی فراهم نشده؟

**Comment by ayub-kokabi** [comment]
چقدر پیگیر! 😊
متأسفانه هر چی جلوتر رفتیم کارها بیشتر شد.
با این حال همچنان تلاشم رو می‌کنم.

**Comment by rayman3003** [comment]
> چقدر پیگیر! 😊
> متأسفانه هر چی جلوتر رفتیم کارها بیشتر شد.
> با این حال همچنان تلاشم رو می‌کنم.

تشکر. چون این قابلیت خیلی برام مهمه, برای همین همچنان پیگیرش هستم. 😅

تشکر از شما. 🤝

**Comment by rayman3003** [comment]
چون ماهِ پیش گیت رو یه آپدیتِ ریز زدید, گفتم شاید فرصتی فراهم شده باشه

**Comment by ayub-kokabi** [comment]
سلام.

خب، دوباره دوستِ پیگیر ما ظاهر شد. 😁

اون آپدیت کوچیکی که دیده بودید فقط مربوط به نام کاربری گیت‌هاب بود و ارتباطی به خود پروژه نداشت.

راستش تو این مدت تنها کاری که تونستم بکنم به‌روزرسانی [[نسخهٔ وب واژه‌دان](https://vajehdan.ir/)](https://vajehdan.ir/) بود. مدت‌ها بود باید یه دستی بهش می‌کشیدم و اشعار فارسی رو هم بهش اضافه می‌کردم. خوشبختانه بالاخره یه فرصت پیدا کردم و تونستم انجامش بدم. ضمناً بخش جستجوی اشعارش خیلی بهتر از نسخهٔ دسکتاپ هست. امتحان کنید.

پیشنهاد شما هم تو ذهنم بود که بهش اضافه کنم. از نظر فنی هم پیاده‌سازی اون در نسخهٔ وب حتی از نسخهٔ دسکتاپ کم‌هزینه‌تره. اما مشکل اصلی اینه که چنین قابلیتی حجم پردازش و هزینهٔ جستجو رو به‌طور قابل‌توجهی افزایش می‌ده. از اونجایی که سایت درآمدی نداره و روی یک سرور نسبتاً ضعیف با منابع محدود میزبانی می‌شه، فعلاً امکان اضافه کردن چنین پردازش‌هایی وجود نداره.

در مورد نسخهٔ دسکتاپ هم، متأسفانه هرچه زمان می‌گذره احتمال اینکه دوباره سراغش برم کمتر می‌شه. این پروژه واقعاً تبدیل به یک پروژهٔ «زیرخاکی» شده؛ با نسخه‌های قدیمی سی‌شارپ نوشته شده و برای توسعهٔ دوباره‌ش باید بخش قابل‌توجهی از کدها به نسخه‌های جدید منتقل بشن. از طرفی سال‌هاست برنامه‌نویسی دسکتاپ رو کنار گذاشتم. اگر بخوام فقط یک تغییر نسبتاً ساده ایجاد کنم، باید اول حدود ۲۰ گیگابایت ویژوال استودیو نصب کنم، بعد دوباره محیط و نسخه‌های جدید سی‌شارپ رو مرور کنم، کدها رو به‌روز کنم و تازه بعد از همهٔ این‌ها برسم به پیاده‌سازی خود قابلیت.

از طرف دیگه، تعداد کاربران نسخهٔ ویندوز خیلی کمه. اینو می‌شه از تعداد Issueها و ستاره‌های گیت‌هاب فهمید. با در نظر گرفتن همهٔ این شرایط، بعید می‌دونم بتونم زمان قابل‌توجهی برای توسعهٔ این پروژه اختصاص بدم.

الان هم باید امیدوار باشیم که یک برنامه‌نویس ویندوزی که با سی‌شارپ کار می‌کنه توسعهٔ نسخهٔ دسکتاپ رو ادامه بده. پروژه هم به همین دلیل آرشیو نکردم.

در کل پروژه‌های متن‌باز همین‌طوری جلو می‌رن. معمولاً یک نفر نمی‌تونه تا ابد چندین پروژه رو هم‌زمان پیش ببره. یکی پروژه‌ای رو شروع می‌کنه، تا جایی که از دستش برمیاد براش وقت می‌ذاره و بعد نوبت می‌رسه به بقیه که ادامه‌ش بدن. شبیه یک امانت یا مشعل که دست‌به‌دست می‌شه.

من هم توی نوبت خودم واقعاً زمان و انرژی زیادی برای واژه‌دان گذاشم و خوشحالم که هنوز داره استفاده می‌شه. الان هم درگیر پروژه‌ها و ایده‌های جدیدی هستم که امیدوارم اون‌ها هم به اندازهٔ واژه‌دان مفید و ماندگار باشن.

امیدوارم واژه‌دان هم با کمک دیگران به مسیرش ادامه بده و هر روز بهتر از قبل بشه.

**Comment by rayman3003** [comment]
> امیدوارم واژه‌دان هم با کمک دیگران به مسیرش ادامه بده و هر روز بهتر از قبل بشه.

بله. کاملا متوجه ی منظورِ شما هستم. خودم پروژه های زیادیو تو گیتهاب دنبال میکنم؛ و همونطور که گفتید, ادامه دادنِ یه پروژه بصورتِ تکنفره, و برای یه مدتِ طولانی, کارِ آسونی نیست.

حیف که من برنامه نویسی بلد نیستم, وگرنه تو این پروژه کمک میکردم؛ و حیف که برنامه نویسِ دیگه ای هم تاکنون پیدا نشده که به این پروژه بپیونده.

شاید ماه ها و سالهای بعد, شخصی پیدا شد و تمایل پیدا کرد به ادامه دادنِ این پروژه. (و اگه مخالفتی ندارید, پیشنهاد میکنم که این ایشو رو نبندید, و همچنان باز بمونه, شاید شخصی بیاد و ببینه, و همکاری کنه در آینده)

دلیلِ اینکه به نسخه ی دسکتاپِ این پروژه بیشتر تمایل دارم, و به نسخه ی وب تمایلی ندارم, اینه که بعد از سالها کارِ آرشیوری کردن, برام ثابت شده که هیچ وبسایتی ماندگار نیست؛ ولی نسخه های دسکتاپ, میتونن همیشه مانا بمونن. 😆

از زحماتی که تا الان برای این پروژه انجام دادید, کمال تشکر رو دارم 🤝 پیروز باشید در مسیرِ پروژه های بعدیتون. ✌️
